### ISSUE TITLE
false negative when indexing an array that is followed by instanceof

### ISSUE THREAD

**Issue by theosotr** [typetools/checker-framework#7341]
### Command

```
javac -J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
-processor org.checkerframework.checker.nullness.NullnessChecker \
-cp checker-framework/checker/dist/checker.jar -Aignorejdkastub Test.java

```

### Program

```java
public class Test {
    public static void main(String[] args) {
        Object[] x = new Object[]{"a"};
        Integer index = null;
        if (x[index] instanceof String) {}
    }
}
```

### Actual behavior

The program passes the checks. At runtime, we get:

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because "<local2>" is null
        at Test.main(Test.java:5)

```

### Expected behavior

An error like:

```
Test.java:5: error: [unboxing.of.nullable] unboxing a possibly-null reference index
        if (x[index] instanceOf String) {}
              ^
1 error
```
